### DISCUSSION TITLE
试用Flet过程中，我有一个关键痛点，急需解决

### DISCUSSION THREAD

**Discussion by LegendaryPistachio** [flet-dev/flet#4228] [Ideas]
flet build apk命令式打包安卓成功率很低，依赖多，遭受多次挫折，有着切肤之痛。许多小伙伴都因为此放弃了使用flet框架搞开发，这非常不利于flet的推广普及。我和我的朋友强烈建议：flet能推出一个flet打包安卓应用的本地打包工具，而且图形化界面，简单好用称手那种！！
造成flet打包安卓失败的原因主要有这么几点：依赖高速网络，中国大陆广大flet爱好者，都使用的是普通网络，网络网速慢，还不稳定。flet打包安卓应用依赖github仓库，我们这的网络经常和github联络超时，这是打包失败次数最多的原因。另外，flet打包安卓应用需要flutter doctor需要的名目繁多的各种依赖，这些资源东一个西一个不集中，也因为没有高速网络原因，还因为谷歌资源都是靠中国大陆的国内镜像，很难短时间凑齐。flet搞开发简单好用，界面元素天生丽质，控件高性能，几分钟就能实现开发构想，这是我和小伙伴们对flet框架趋之若鹜的根本原因，但是flet build apk命令式打包安卓成功率很低，依赖多，没有本地打包工具，而且图形化界面，简单好用称手那种！！我和我的小伙伴很焦躁。这严重影响到开发效率，拖了后腿，体验很差。解决这个问题是当务之急，中国大陆大概有四亿python爱好者，也是flet框架使用者，这是一股不可小觑的力量，希望flet格外关注这些人群，为这些人搞好服务。如果能这样，Flet框架出类拔萃的优秀特性，让flet框架在中国大陆、在全球前端开发中脱颖而出，如火如荼，指日可待！

**Comment by FeodorFitsner** [comment]
Translated to English:

**During the trial of Flet, I had a key pain point that needed to be solved urgently**

The success rate of flet build apk command packaging Android is very low, with a lot of dependence, suffering from many setbacks, and the pain of cutting the skin. Because of this, many friends have given up using the flet framework for development, which is very detrimental to the popularization of flet. My friends and I strongly recommend: flet can launch a local packaging tool for flet packaging Android applications, and the graphical interface is simple and easy to use!!

The main reasons for the failure of flet packaging Android are as follows: relying on high-speed networks, the majority of flet enthusiasts in mainland China are using ordinary networks, and the network speed is slow and unstable. Flet packaging Android applications rely on the github repository. Our network often times out with github, which is the reason for the most packaging failures. In addition, flet packaging Android applications requires a variety of dependencies required by flutter doctor. These resources are not concentrated one by one, and because there is no high-speed network, and because Google resources all rely on domestic mirror images in mainland China, it is difficult to collect them in a short time. Flet is simple and easy to develop. The interface elements are naturally beautiful, and the controls are high-performance. The development concept can be realized in a few minutes. This is the fundamental reason why my friends and I flock to the flet framework. However, the success rate of flet build apk command packaging Android is very low, and there is a lot of dependence, and there is no The packaging tool, and the graphical interface, is simple and easy to use!! My friend and I are very anxious. This seriously affects the development efficiency, drags down, and the experience is very poor. It is urgent to solve this problem. There are about 400 million python enthusiasts in mainland China who are also users of the flet framework. This is a force that cannot be underestimated. I hope flet will pay special attention to these people and provide good services for these people. If so, the outstanding features of the Fleet framework will make the fleet framework stand out in mainland China and global front-end development. It is in full swing and just around the corner!

**Reply by LegendaryPistachio** [reply]
flet急迫需要一个本地化打包安卓应用工具，其次才是图形化界面

**Reply by Assassinator-567** [reply]
That's what I'm trying to say, it's so hard to pack, especially in China

**Comment by Dexter-DS** [comment]
The process of building APKs for Android using Flet’s command-line interface has proven to be unreliable and frustrating due to its numerous dependencies, causing many developers, including myself, to reconsider using the Flet framework. This is a significant hindrance to its wider adoption. My colleagues and I strongly recommend that Flet develops a local packaging tool for Android apps, complete with a user-friendly, graphical interface. The primary challenges faced during Android packaging with Flet include the need for a high-speed internet connection, which is not commonly available in mainland China, leading to frequent timeouts with GitHub repositories – the main culprit for packaging failures. Moreover, the tool requires a myriad of dependencies for ‘flutter doctor’, which are scattered and hard to obtain due to limited internet speeds and the reliance on domestic mirrors for Google resources in China. Despite Flet’s ease of use, aesthetic interface elements, and high-performance controls, which attract developers like us, the current state of Android packaging is a major setback. It significantly impacts our productivity and user experience. Addressing this issue is urgent, especially considering the large Python community in China, which is also a potential user base for Flet. By catering to this demographic, Flet can enhance its presence in China and globally, becoming a leading choice in front-end development.

**Comment by YueXiaWyl** [comment]
I recently used flet to develop an app. I have to say it is really easy to use. The computer and mobile interfaces are adaptive, which reduces a lot of development. It is very good. When I finished the development, I planned to package it into apk and use it on the phone. According to the official packaging document, either git pull failed or other dependencies were missing. It took half a month and it has not been packaged yet. I need a good official solution to solve the problem of local packaging difficulties.

**Reply by ndonkoHenri** [reply]
Perhaps make use of Github actions?: https://youtu.be/ObO-D2TD_wo

**Comment by LegendaryPistachio** [comment]
The Flet framework needs to roll out a tool for packaging Android apps locally as soon as possible.
Flet框架需要尽快推出一个本地打包安卓应用的工具

**Comment by kiriyama0133** [comment]
亲爱的作者，我想问一下处于conda的环境中使用flet build进行构建Windows app的时候，会不会造成flet无法正确找到依赖 造成应用无法正常打开

**Comment by SystemFileB** [comment]
🤔或者你可以试试github actions来构建，国外的服务器，我看日志都打包好了  
怎么回事看我的all-of-resources仓库
